### ISSUE TITLE
investigate and improve apple silicon performance

### ISSUE THREAD

**Issue by Trivo25** [o1-labs/snarkyjs#719]
during the last office hours, we have received complains about degraded (or rather no improvement at all compared to before the original fix https://github.com/o1-labs/snarkyjs/pull/683) apple silicon performance in snarkyjs 0.8.0

another nice-to-have could be benchmarks that users can run, potentially baked in to the zk cli

**Comment by jasongitmail** [comment]
@Trivo25 If you're able to contact these devs, can you ask them to provide the output from `zk system` please? To add to this issue

**Comment by maht0rz** [comment]
here's my `zk system`:

```
✗ zk system
Please include the following when submitting a Github issue:

  System:
    OS: macOS 13.0
    CPU: (10) arm64 Apple M1 Max
  Binaries:
    Node: 17.8.0 - ~/.nvm/versions/node/v17.8.0/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v17.8.0/bin/npm
  npmPackages:
    snarkyjs: ^0.8.0 => 0.8.0 
  npmGlobalPackages:
    zkapp-cli: 0.5.4
```


We've tried proving worlde or some other zkApp with Jack at the last office hours and we found out the performance was still degraded.

**Comment by Trivo25** [comment]
seems like bindings got overwritten

**Comment by mitschabaude** [comment]
hm, I would like to understand this. in `main`, the changes are in the bindings -- search for `get_efficient_num_workers` in this file: https://raw.githubusercontent.com/o1-labs/snarkyjs/main/src/node_bindings/snarky_js_node.bc.cjs

**Comment by Trivo25** [comment]
If i am not mistaken the bindings weren't up to date with the fix with the state from release 0.8.0 (which they should have) - but they are now https://raw.githubusercontent.com/o1-labs/snarkyjs/c5a36207a7d345d4b96038569d7db66101ec07cd/src/node_bindings/snarky_js_node.bc.cjs so it seems like a PR remove the fix and another one after that re-added the change - probably some out of sync work on the Mina side

**Comment by Trivo25** [comment]
Just confirmed my assumption. Tested by looking for the added code snippet in the bindings `get_efficient_num_workers` as well as running a handful of tests on my M2 from release tag 0.8.0 and main - version 0.8.0 shipped without the fix but the next update will have the updated bindings
